### PR TITLE
CB-17100: Trigger restore on datalake resize when RAZ is enabled.

### PR DESCRIPTION
**Pull request by kkalvagadda1**
See detailed description in the commit message.

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeFlowEventChainFactory.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeFlowChain
         // Create new
         chain.add(new SdxEvent(SDX_VALIDATION_EVENT.event(), event.getResourceId(), event.getSdxCluster().getClusterName(), event.getUserId()));
 
-        if (event.shouldTakeBackup() && !event.getSdxCluster().isRangerRazEnabled()) {
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeFlowEventChainFactory.java:62]: In the past, we used the same input to decide if the backup or restore has to be triggered as part of resizing.

With the new change that was introduced, we have different validations for backup or restore operations. This change is used for validations done for restoration.

[the remark this answers is not in the document]
Comment (sxxgrc) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeFlowEventChainFactory.java:62]: Okay sounds good.

**Review by sxxgrc — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/DistroxService.java [hunk 2]
```diff
@@ -65,7 +65,14 @@ public void restartAttachedDistroxClusters(String envCrn) {
         // We allow the refresh of DHs in NODE_FAILURE states to "fix" clusters that end up in this state due to their
         // services being unable to communicate with the DL due to a migration process or some other flow.
         Collection<StackViewV4Response> availableClusters = attachedDistroXClusters.stream()
-                .filter(cluster -> cluster.getStatus() == Status.AVAILABLE || cluster.getStatus() == Status.NODE_FAILURE)
+                .filter(cluster ->  {
+                    if (cluster.getStatus() == Status.AVAILABLE || cluster.getStatus() == Status.NODE_FAILURE) {
+                        return true;
+                    } else {
+                        LOGGER.info("Skipping restart for DH cluster: {}, Status: {}", cluster.getName(), cluster.getStatus().name());
```
Comment (horadla23) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/DistroxService.java:72]: if we want to log all related DHs which cannot be restarted, I would collect them and log them into one log line using DEBUG level

**Review by horadla23 — COMMENTED** [review]
one suggestions about logging, otherwise lgtm

**Review by RokolyA94 — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/DistroxService.java (hunk 2 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/DistroxService.java:72]: We need to log the cluster name and their status. 
I prefer to have them logged separately as we can not log all of them cleanly in one line.

If it were just the names, one logline would be fine.

**Review by horadla23 — APPROVED** [review]

**Pull request merged**